go/interpreter/lfvm/memory.go

```diff
@@ -188,35 +180,19 @@ func (m *Memory) SetWithCapacityAndGasCheck(offset, size uint64, value []byte, c
 	if err != nil {
 		return err
 	}
-	err = m.Set(offset, size, value)
-	if err != nil {
-		return err
-	}
-	return nil
+	return m.Set(offset, size, value)
 }
 
 func (m *Memory) SetWithCapacityCheck(offset, size uint64, value []byte) error {
 	m.EnsureCapacityWithoutGas(size)
-	if size > 0 {
-		if offset+size < offset {
-			return errGasUintOverflow
-		}
-		if offset+size > m.Len() {
-			return fmt.Errorf("memory too small, size %d, attempted to write %d bytes at %d", m.Len(), size, offset)
-		}
-		copy(m.store[offset:offset+size], value)
-	}
-	return nil
+	return m.Set(offset, size, value)
```

Comment:
it could be the caller's responsibility to call it with `offset+size`, but I don't like that approach. this does seem weird, specially because whenever `EnsureCapacityWithoutGas` is called, the memory operation that comes after always have offset and size. 
That said, this change should probably be in a different PR with a little more analysis.

Reply:
It looks like there is only a single call of this function which indeed uses the offset `0`. It should be checked whether this function can be eliminated in a follow-up PR.